sources/web/datalab/polymer/components/datalab-data/datalab-data.ts

```diff
@@ -18,208 +18,170 @@
-  // private _collectionsFilterValue : string;   //TODO(jimmc): uncomment when ready to use
-
-  /** The value the user can enter to filter the tables. */
-  // private _tablesFilterValue : string;    //TODO(jimmc): uncomment when ready to use
-
-  private _collectionsList: Collection[];
-  private _tablesList: Table[];
+  private _isDetailsPaneToggledOn: boolean;
+  private _resultsList: Result[];
+  private _searchValue: string;
 
   static get is() { return 'datalab-data'; }
 
   static get properties() {
     return {
-      _collectionsFilterValue: {
-        type: String,
-        value: 'no-filter',
+      _isDetailsPaneToggledOn: {
+        type: Boolean,
+        value: true,
       },
-      _collectionsList: {
-        observer: '_renderCollectionsList',
+      _resultsList: {
+        observer: '_renderResultsList',
         type: Array,
         value: () => [],
       },
-      _tablesFilterValue: {
+      _searchValue: {
         type: String,
-        value: 'no-filter',
-      },
-      _tablesList: {
-        observer: '_renderTablesList',
-        type: Array,
-        value: () => [],
+        value: '',
       },
     };
   }
 
   ready() {
     super.ready();
 
-    this._collectionsList = [];
-    const collectionsElement = this.$.collections as ItemListElement;
-    if (collectionsElement) {
-      collectionsElement.addEventListener('itemDoubleClick',
-                                    this._collectionsDoubleClicked.bind(this));
-      collectionsElement.addEventListener('selected-indices-changed',
-                                    this._collectionsSelectionChanged.bind(this));
+    this._resultsList = [];
+    const resultsElement = this.$.results as ItemListElement;
+    if (resultsElement) {
+      resultsElement.addEventListener('itemDoubleClick',
+                                    this._resultsDoubleClicked.bind(this));
+      resultsElement.addEventListener('selected-indices-changed',
+                                    this._resultsSelectionChanged.bind(this));
     }
 
-    this._tablesList = [];
-    const tablesElement = this.$.tables as ItemListElement;
-    if (tablesElement) {
-      tablesElement.addEventListener('itemDoubleClick',
-                                    this._tablesDoubleClicked.bind(this));
-      tablesElement.addEventListener('selected-indices-changed',
-                                    this._tablesSelectionChanged.bind(this));
-    }
+    (this.$.results as ItemListElement).columns = ['Name', 'Type'];
+    this.$.searchKeys.target = this.$.searchBox;
   }
 
-  _generateFakeCollectionsListForTesting() {
-    const collectionsList = [];
-    const count = Math.floor(Math.random() * 20);
-    for (let i = 0; i < count; i++) {
-      collectionsList.push(this._generateFakeCollectionForTesting());
-    }
-    return collectionsList;
+  /** Sends the user's query to the search API, renders results as they get returned. */
+  _search() {
+    this._resultsList = [];
```

Comment:
Right now the search API is pretty bad: I am calling three separate APIs that return paged results, with different kinds of filtering (or none), that are responding at different speeds. Once I figure out how to make that work better - or switch to something completely different - then I can revisit this issue. Meanwhile, I added a TODO.